### ISSUE TITLE
Missing rates for Switzerland (CH)

### ISSUE THREAD

**Issue by leehambley** [adamcooke/vat-rates#24]
I see they are also missing from the page at http://ec.europa.eu/taxation_customs/taxation/vat/traders/vat_community/index_en.htm but I wonder if they should be included in the cache/jsonvat.com output somewhere?

**Comment by stayhero** [comment]
Switzerland is not an EU country. Of course it would be nice to have a source for all countries and their tax rates worldwide, but I guess that'd be too much work. And the use case for the vat-rates gem is probably more targeted to customers selling/billing within the EU.

**Comment by leehambley** [comment]
Good point :) I'm working with a German company who consider [D-A-CH](https://de.wikipedia.org/wiki/D-A-CH) as their primary user base. Thanks for clearing that up for me, though. I wonder how best I can solve this :)

**Comment by stayhero** [comment]
@leehambley  It depends. :-) From which country does your customer do their services or sell the products? If they e.g. sell from Germany they do not have to add any VAT taxes for shipments to Switzerland. As Switzerland is considered a third-party county (just like you would not add any taxes when you ship something from Germany to US).

It's rather the buyer who is responsible for paying customs and his local taxes (importation VAT). It gets more complicated if they have their own warehouse in Switzerland (in that case you'd probably need to know the Swiss VAT rates)... but I guess your employer/customer already knows what taxes should be applied in such cases. Ask him :)

**Comment by leehambley** [comment]
> @leehambley It depends. :-) From which country does your customer do their services or sell the products? If they e.g. sell from Germany they do not have to add any VAT taxes for shipments to Switzerland. As Switzerland is considered a third-party country (just like you would not add any taxes when you ship something from Germany to US).

We have some deal with Switzerland as far as I understand, as we do a lot of business there, we are somehow responsible for VAT in that country through some deal that exists between Germany and Switzerland for collaborative collection of tax.

I only know that I have a table of "given… should" rules, and some of the branches deal with CH customers, so I'm just trying to get to a green test state :)

**Comment by stayhero** [comment]
@leehambley  I would give them a backend interface where they can update the VAT rates by themselves (just remember to support setting VAT rates for the future). 

Or you could offer them integration with something like https://www.taxamo.com/ if they really sell worldwide and have branches in a lot of countries. :-)

**Comment by leehambley** [comment]
Unfortunately the 3rd party integration is not allowed, and thus we chose your gem for caching, but agreed that we're probably doing something a bit wrong. This stuff is horribly complicated. My own company uses Braintree+Taxamo and it works beautifully.

**Comment by stayhero** [comment]
It's not my gem... I just saw your issue and wanted to answer. Anyways I doubt @adamcooke  will implement swiss VAT rates because it targets EU countries. 

I know it's complicated. This gem is also not 100% satisfying. The VAT rates also differ not only per country but also between the kind of product. E.g. an ebook sold in France has the reduced-1 VAT rate (5,5%) whereas in Germany it has the standard (19%) rate.

I don't know what your client sells exactly but I'm afraid if you really want to be on the safe side, your client should supply the vat rates. For our purposes (invoicing SaaS solution) we use the VAT rates gem to show sample defaults but ultimatively the customer is responsible for entering the correct VAT rate.

I think you could - with a lot of work - extend this gem to be much more usable for these kind of use cases. Having all VAT rates in the EU is a good start. But you'd also need a list of products where different VAT rates applies in the different EU countries. The users of this gem could then provide a list of "special" products which are considered when the VAT rates for the countries are calculated.

E.g. 
```ruby
  special_products = VATRates.special_product_types(locale: :en)
  # use this list to let the user deceide which kind of VAT rate
  # is applicable for the desired product, like either :standard, :reduced, :discounted
  # or a product type (e.g. ebook). 
  # This could be used to fill a select-option field
```

Basically we'd need a localizable list of product types (for types, where special VAT consideration is necessary). The VAT calculation then could be done like

```ruby
  vat_percentage = VATRates.get(country: :de, type: :ebooks)
  # or 
  vat_percentage = VATRates.get(country: :de, type: :standard)
```

I'm not sure if anybode else finds this useful. I would offer my help compiling such a list. But I'm not sure if it's too much of a hassle to maintain such a list.

**Comment by driade** [comment]
@stayhero @adamcooke I find it useful, I do it on my monthly basics for a site of mine (lektu.com). I would be glad to provide the current VATs for ebooks as well as to maintain the data.

**Comment by driade** [comment]
Abrir un repo propio con los valores?
